# **Lab Work 1**

## STEP 1 - Load Data

In [12]:
#STEP 1  Load Data
import pandas as pd

#import Data (i've uploaded the Datasets into google colab)
data = "/content/Titanic-Dataset.csv" #data_path
df = pd.read_csv(data) # load dataset

df.head() #show 5 first data (Test)





PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

## STEP 2 - Data Checking

In [15]:
#STEP 2 Data Checking

df.info() # You can get general information about the data

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


From df.info() we get the information about the number of non null data for each column. Dispite of that, we also know that,

Age type is float64

Fare type is float64

Embarked type is object.

In [16]:
df.isnull().sum() # we will check the total null data from these columns


PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

From that information, we know that

The total of null values of Age is 177.

The total of null values of Cabin is 687.

The total of null values of Embarked is 2.

## STEP 3 - Imputation

In [18]:
#Step 3 - Imputation
# Age - mean
df['Age'].fillna(value=df['Age'].mean(), inplace=True)

# Cabin - "DECK"
df['Cabin'].fillna(value="DECK", inplace=True)

# Embarked - modus
df['Embarked'].fillna(value=df['Embarked'].mode, inplace=True)

perform imputation for missing data in "Age," "Cabin," and "Embarked." The strategy we will use is as follows,

"Age" --> Since "Age" is nominal data, we will use the mean strategy.

"Cabin" --> "Cabin" is related to passenger cabin numbers. Here, we will assume that all passengers without cabin numbers stayed on the ship's decks. Therefore, we will substitute missing data with the information "DECK."

"Embarked" --> "Embarked" represents the location of passenger embarkation. There are only 2 missing data points here. Since the data is nominal, we can use the mode to substitute missing data.


## STEP 4 - Verification

In [19]:
#Step 4 - Verification
df.info()

#Perform a recheck Furthermore, we can also verify whether the filled values are correct by directly checking the data. The results of the imputation process are as follows,

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          891 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        891 non-null    object 
 11  Embarked     891 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [21]:
#Extracting The data
df.to_csv('/content/Titanic-Dataset-fixed.csv', index=False)

# **Lab Work 2**

Feature Extraction
In this stage, we will select which variables we will use as features. In the foundational theory section, we have already learned that not all variables can always be used as features. Therefore, we can choose features that are deemed important for the purpose of creating a machine learning model. We will use the same case, which is to identify the characteristics of Titanic passengers who survived. Thus, the features we will use are,

"Survived"

"Pclass"

"Sex"

"Age"

"Cabin"

We will separate these variables from the others. The data we will use is the Titanic data that has been repaired through the imputation process.

## Step 0 - Load Library

In [7]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder, StandardScaler

## Step 1 - Load Data

In [22]:
dpath = '/content/Titanic-Dataset-fixed.csv'
df = pd.read_csv(dpath)
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500  DECK        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250  DECK        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500  DECK        S

## Step 2 - Slice The Data

In [8]:
#We will select only the columns that we think is features.
df = df[['Survived', 'Pclass', 'Age', 'Sex', 'Cabin']]
df.head()

Survived  Pclass   Age     Sex Cabin
0         0       3  22.0    male  DECK
1         1       1  38.0  female   C85
2         1       3  26.0  female  DECK
3         1       1  35.0  female  C123
4         0       3  35.0    male  DECK

## Step 3 - Encoding

We will perform encoding on categorical data, namely "Sex" and "Cabin." This step is necessary because machine learning algorithms perform mathematical calculations. The strategy we will use is Label Encoding. Label Encoding is similar to Ordinal Encoding, with the only difference being the sorting process. Label Encoding does not sort the data beforehand.

In [23]:
le = LabelEncoder() # initiate LabelEncoder object
df['Sex'] = le.fit_transform(df['Sex']) # encoding
df['Cabin'] = le.fit_transform(df['Cabin']) # encoding

## Step 4 - Verification

In [24]:
#rerun df.head command to verify the result.
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex   Age  SibSp  Parch  \
0                            Braund, Mr. Owen Harris    1  22.0      1      0   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  38.0      1      0   
2                             Heikkinen, Miss. Laina    0  26.0      0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  35.0      1      0   
4                           Allen, Mr. William Henry    1  35.0      0      0   

             Ticket     Fare  Cabin Embarked  
0         A/5 21171   7.2500    115        S  
1          PC 17599  71.2833     81        C  
2  STON/O2. 3101282   7.9250    115        S  
3            113803  53.1000     55        S  
4            373450   8.0500    115        S

## Step 5 - Standardization

In this stage, we will standardize "Age." This is not actually necessary because the values in "Age" are not too far apart. The purpose of this step is to demonstrate the standardization process.

In [25]:
std = StandardScaler()
df['Age'] = std.fit_transform(df[['Age']])

## Step 6 - Verification

In [26]:
#run df.head to verufy the result
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex       Age  SibSp  \
0                            Braund, Mr. Owen Harris    1 -0.592481      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  0.638789      1   
2                             Heikkinen, Miss. Laina    0 -0.284663      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  0.407926      1   
4                           Allen, Mr. William Henry    1  0.407926      0   

   Parch            Ticket     Fare  Cabin Embarked  
0      0         A/5 21171   7.2500    115        S  
1      0          PC 17599  71.2833     81        C  
2      0  STON/O2. 3101282   7.9250    115        S  
3      0            113803  53.1000     55        S  
4      0            373450   8.0500    115        S

In [28]:
#Extract Data
df.to_csv('/content/Titanic-Dataset-selected.csv', index=False)

# **Lab Work 3**

In this lab work, we will try three different strategies for creating training, validation, and test data. The data used will be the Titanic ship data for which we extracted features in the previous practical session.

## Random Split

### Step 1 - Load Data

In [30]:
import pandas as pd

df = pd.read_csv('/content/Titanic-Dataset-selected.csv')
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex       Age  SibSp  \
0                            Braund, Mr. Owen Harris    1 -0.592481      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  0.638789      1   
2                             Heikkinen, Miss. Laina    0 -0.284663      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  0.407926      1   
4                           Allen, Mr. William Henry    1  0.407926      0   

   Parch            Ticket     Fare  Cabin Embarked  
0      0         A/5 21171   7.2500    115        S  
1      0          PC 17599  71.2833     81        C  
2      0  STON/O2. 3101282   7.9250    115        S  
3      0            113803  53.1000     55        S  
4      0            373450   8.0500    115        S

### Step 2 - Split Data

In [31]:
# Split data
from sklearn.model_selection import train_test_split

# Split data training dan dan lainnya
# data lainnya, akan kita split lagi menjadi validasi dan testing.
# Rasio yang akan kita gunakan adalah 8:1:1
df_train, df_unseen = train_test_split(df, test_size=0.2, random_state=0)

# Split lagi antara validasi dan testing
df_val, df_test = train_test_split(df_unseen, test_size=0.5, random_state=0)

# Cek masing-masing ukuran data

print(f'Jumlah data asli: {df.shape[0]}')
print(f'Jumlah data train: {df_train.shape[0]}')
print(f'Jumlah data val: {df_val.shape[0]}')
print(f'Jumlah data test: {df_test.shape[0]}')

# Cek rasio tiap label
print('=========')
print(f'Jumlah label data asli:\n{df.Survived.value_counts()}')
print(f'Jumlah label data train:\n{df_train.Survived.value_counts()}')
print(f'Jumlah label data val:\n{df_val.Survived.value_counts()}')
print(f'Jumlah label data test:\n{df_test.Survived.value_counts()}')

Jumlah data asli: 891
Jumlah data train: 712
Jumlah data val: 89
Jumlah data test: 90
Jumlah label data asli:
Survived
0    549
1    342
Name: count, dtype: int64
Jumlah label data train:
Survived
0    439
1    273
Name: count, dtype: int64
Jumlah label data val:
Survived
0    53
1    36
Name: count, dtype: int64
Jumlah label data test:
Survived
0    57
1    33
Name: count, dtype: int64


## Stratified Split

### Step 1 - Load Data

In [32]:
import pandas as pd

df2 = pd.read_csv('/content/Titanic-Dataset-selected.csv')
df2.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex       Age  SibSp  \
0                            Braund, Mr. Owen Harris    1 -0.592481      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  0.638789      1   
2                             Heikkinen, Miss. Laina    0 -0.284663      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  0.407926      1   
4                           Allen, Mr. William Henry    1  0.407926      0   

   Parch            Ticket     Fare  Cabin Embarked  
0      0         A/5 21171   7.2500    115        S  
1      0          PC 17599  71.2833     81        C  
2      0  STON/O2. 3101282   7.9250    115        S  
3      0            113803  53.1000     55        S  
4      0            373450   8.0500    115        S

### Step 2 - Split Data

In [33]:
# Split data
from sklearn.model_selection import train_test_split

# Split data training dan dan lainnya
# data lainnya, akan kita split lagi menjadi validasi dan testing.
# Rasio yang akan kita gunakan adalah 8:1:1
df2_train, df2_unseen = train_test_split(df2, test_size=0.2, random_state=0, stratify=df['Survived'])

# Split lagi antara validasi dan testing
df2_val, df2_test = train_test_split(df2_unseen, test_size=0.5, random_state=0, stratify=df_unseen['Survived'])

# Cek masing-masing ukuran data

print(f'Jumlah label data asli:\n{df2.Survived.value_counts()}')
print(f'Jumlah label data train:\n{df2_train.Survived.value_counts()}')
print(f'Jumlah label data val:\n{df2_val.Survived.value_counts()}')
print(f'Jumlah label data test:\n{df2_test.Survived.value_counts()}')

Jumlah label data asli:
Survived
0    549
1    342
Name: count, dtype: int64
Jumlah label data train:
Survived
0    439
1    273
Name: count, dtype: int64
Jumlah label data val:
Survived
0    51
1    38
Name: count, dtype: int64
Jumlah label data test:
Survived
0    59
1    31
Name: count, dtype: int64


## Cross Validation 1

### Step 1 - Load Data

In [35]:
import pandas as pd

df3 = pd.read_csv('/content/Titanic-Dataset-selected.csv')
df3.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex       Age  SibSp  \
0                            Braund, Mr. Owen Harris    1 -0.592481      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  0.638789      1   
2                             Heikkinen, Miss. Laina    0 -0.284663      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  0.407926      1   
4                           Allen, Mr. William Henry    1  0.407926      0   

   Parch            Ticket     Fare  Cabin Embarked  
0      0         A/5 21171   7.2500    115        S  
1      0          PC 17599  71.2833     81        C  
2      0  STON/O2. 3101282   7.9250    115        S  
3      0            113803  53.1000     55        S  
4      0            373450   8.0500    115        S

### Step 2 - Split Data

In [36]:
# Implementasi k-fold cross validation (random) dengan training dan testing saja
from sklearn.model_selection import KFold

# inisiasi obyek kfold
kf = KFold(n_splits=4)
print(f'Jumlah fold: {kf.get_n_splits()}')
print(f'Obyek KFold: {kf}')

# Lakukan splitting dengan KFold
kf_split = kf.split(df3)
print(f'Jumlah data df: {df.shape[0]}')

# cek index data tiap fold
for train_index, test_index in kf_split:
    print(f'Index train: {train_index} | Index test: {test_index}')

Jumlah fold: 4
Obyek KFold: KFold(n_splits=4, random_state=None, shuffle=False)
Jumlah data df: 891
Index train: [223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240
 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258
 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276
 277 278 279 280 281 282 283 284 285 286 287 288 289 290 291 292 293 294
 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312
 313 314 315 316 317 318 319 320 321 322 323 324 325 326 327 328 329 330
 331 332 333 334 335 336 337 338 339 340 341 342 343 344 345 346 347 348
 349 350 351 352 353 354 355 356 357 358 359 360 361 362 363 364 365 366
 367 368 369 370 371 372 373 374 375 376 377 378 379 380 381 382 383 384
 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 400 401 402
 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 419 420
 421 422 423 424 425 426 427 428 429 430 431 432 433 434 435 436 437 438
 439 440 44

## Cross Validation 2

In this work, we are taking a slightly different approach compared to the previous work. First, we will split the data into training and test data. Then, we will further split the training data into several parts using cross-validation for training and validation data purposes.

### Step 1 - Load Data

In [38]:
import pandas as pd

df4 = pd.read_csv('/content/Titanic-Dataset-selected.csv')
df4.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name  Sex       Age  SibSp  \
0                            Braund, Mr. Owen Harris    1 -0.592481      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...    0  0.638789      1   
2                             Heikkinen, Miss. Laina    0 -0.284663      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)    0  0.407926      1   
4                           Allen, Mr. William Henry    1  0.407926      0   

   Parch            Ticket     Fare  Cabin Embarked  
0      0         A/5 21171   7.2500    115        S  
1      0          PC 17599  71.2833     81        C  
2      0  STON/O2. 3101282   7.9250    115        S  
3      0            113803  53.1000     55        S  
4      0            373450   8.0500    115        S

### Step 2 - Split Data

In [39]:
# Implementasi k-fold cross validation (random) dengan training, validation, dan testing data
from sklearn.model_selection import train_test_split, KFold

# Split dulu antara data training dan testing dengan train_test_split
# Rasio 8:2 untuk training dan testing
df4_train, df4_test = train_test_split(df4, test_size=0.2, random_state=0)

# inisiasi obyek kfold
kf2 = KFold(n_splits=4)
print(f'Jumlah fold: {kf2.get_n_splits()}')
print(f'Obyek KFold: {kf2}')

# Lakukan splitting dengan KFold untuk data df_training
# Dengan acara ini, kita masih memiliki data testing untuk keperluan pengujian model
# namun tetap dapat melakukan evaluasi dengan menggunakan data validasi
kf2_split = kf2.split(df_train)
print(f'Jumlah data df_train: {df4_train.shape[0]}')

# cek index data tiap fold
for train_index, test_index in kf2_split:
    print(f'Index train: {train_index} | Index test: {test_index}')

Jumlah fold: 4
Obyek KFold: KFold(n_splits=4, random_state=None, shuffle=False)
Jumlah data df_train: 712
Index train: [178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195
 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213
 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231
 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249
 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267
 268 269 270 271 272 273 274 275 276 277 278 279 280 281 282 283 284 285
 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303
 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 320 321
 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339
 340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357
 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375
 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391 392 393
 394 

# **Lab Work 4**

In this work, we will attempt to extract features from an image. The process we will undertake is relatively simple. A comprehensive discussion regarding data preprocessing and feature extraction in image data will be covered in the digital image processing lectures.

## Step 0 - Install Library

In [40]:
#we will use library called Pillow to perform some image processing.
!pip install Pillow

## Step 1 - Load Image

In [46]:
from PIL import Image

img = Image.open('/content/Image/image - lena.png')
img.show() # show the images

## Step 2 - Feature Extraction

In [47]:
# extact red, green, blue, channel
r, g, b = img.split()

# check the length of red feature
print(len(r.histogram()))

# Print the historgram of red channel
print(r.histogram())

256
[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 8, 6, 10, 9, 20, 37, 32, 41, 53, 74, 103, 132, 179, 193, 242, 238, 336, 386, 404, 483, 607, 628, 771, 785, 1001, 1016, 1177, 1269, 1332, 1446, 1487, 1457, 1574, 1585, 1591, 1557, 1569, 1660, 1648, 1420, 1559, 1418, 1384, 1319, 1342, 1156, 1120, 955, 969, 828, 782, 752, 737, 719, 700, 628, 673, 587, 617, 610, 592, 557, 593, 552, 566, 582, 559, 571, 520, 664, 650, 618, 730, 594, 667, 675, 685, 771, 715, 667, 740, 744, 766, 765, 772, 817, 817, 744, 806, 760, 777, 812, 797, 799, 861, 814, 910, 907, 918, 888, 1011, 879, 996, 912, 952, 884, 1074, 977, 1073, 1040, 1216, 1250, 1403, 1534, 1639, 1682, 1776, 1874, 1769, 1582, 1743, 1441, 1477, 1483, 1409, 1437, 1449, 1389, 1479, 1592, 1655, 1657, 1666, 1857, 1896, 1813, 1979, 1814, 1956, 1928, 2055, 2012, 2303, 2333, 2670, 2787, 3232, 3154, 3476, 3424, 3516, 3102, 31

# **LAB ASSIGNMENT**

**Assignment Description**
In this practical assignment, you will be using the "Wisconsin Breast Cancer" dataset. The dataset consists of 569 instances used for diagnosing Malignant (M) and Benign (B) cancer types. Your tasks are as follows:


  Separate the variables that can be used and the ones that cannot be used.

  Perform encoding on the "diagnosis" column.

  Standardize all columns with numeric values.


Perform a stratified split of the data to create training and test data with an 80:20 ratio.

## Separate Variables

In [52]:
#Load Data
import pandas as pd

data = '/content/wbc.csv' # path dataset
df = pd.read_csv(data) # load dataset

#Check data value
# df.head()
#check data information
df.info()

id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  texture_worst  perimeter_worst  area_worst  smoothness_worst  \
0  ...          17.33           184.60      2019.0            0.1622   
1  ...          23.41           158.80      1956.0            0.1238   
2  ...          25.53           152.50      1709.0            0.1444   
3  ...          26.50            98.87       567.7            0.2098   
4  ...          16.67           152.20      1575.0            0.1374   

   compactness_worst  concavity_worst  concave points_worst  symmetry_worst  \
0             0.6656           0.7119                0.2654          0.4601   
1             0.1866           0.2416                0.1860          0.2750   
2             0.4245           0.4504                0.2430          0.3613   
3             0.8663           0.6869                0.2575          0.6638   
4             0.2050           0.4000                0.1625          0.2364   

   fractal_dimension_worst  Unnamed: 32  
0                  0.11890          NaN  
1                  0.08902          NaN  
2                  0.08758          NaN  
3                  0.17300          NaN  
4                  0.07678          NaN  

[5 rows x 33 columns]

We wont be needing id and unnamed:31 so we gonna drop them

In [59]:
#dropping id,unnamed:32
df = df.drop(columns=['id', 'Unnamed: 32'])
df.head()


diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0         M        17.99         10.38          122.80     1001.0   
1         M        20.57         17.77          132.90     1326.0   
2         M        19.69         21.25          130.00     1203.0   
3         M        11.42         20.38           77.58      386.1   
4         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   symmetry_mean  ...  radius_worst  texture_worst  perimeter_worst  \
0         0.2419  ...         25.38          17.33           184.60   
1         0.1812  ...         24.99          23.41           158.80   
2         0.2069  ...         23.57          25.53           152.50   
3         0.2597  ...         14.91          26.50            98.87   
4         0.1809  ...         22.54          16.67           152.20   

   area_worst  smoothness_worst  compactness_worst  concavity_worst  \
0      2019.0            0.1622             0.6656           0.7119   
1      1956.0            0.1238             0.1866           0.2416   
2      1709.0            0.1444             0.4245           0.4504   
3       567.7            0.2098             0.8663           0.6869   
4      1575.0            0.1374             0.2050           0.4000   

   concave points_worst  symmetry_worst  fractal_dimension_worst  
0                0.2654          0.4601                  0.11890  
1                0.1860          0.2750                  0.08902  
2                0.2430          0.3613                  0.08758  
3                0.2575          0.6638                  0.17300  
4                0.1625          0.2364                  0.07678  

[5 rows x 31 columns]

## Encoding 'Diagnosis'

The strategy we will use is Label Encoding

In [62]:
from sklearn.preprocessing import LabelEncoder, StandardScaler
le = LabelEncoder() # initiate LabelEncoder object
df['diagnosis'] = le.fit_transform(df['diagnosis']) # encoding

#check result
df.head(20)

diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0           1        17.99         10.38          122.80     1001.0   
1           1        20.57         17.77          132.90     1326.0   
2           1        19.69         21.25          130.00     1203.0   
3           1        11.42         20.38           77.58      386.1   
4           1        20.29         14.34          135.10     1297.0   
5           1        12.45         15.70           82.57      477.1   
6           1        18.25         19.98          119.60     1040.0   
7           1        13.71         20.83           90.20      577.9   
8           1        13.00         21.82           87.50      519.8   
9           1        12.46         24.04           83.97      475.9   
10          1        16.02         23.24          102.70      797.8   
11          1        15.78         17.89          103.60      781.0   
12          1        19.17         24.80          132.40     1123.0   
13          1        15.85         23.95          103.70      782.7   
14          1        13.73         22.61           93.60      578.3   
15          1        14.54         27.54           96.73      658.8   
16          1        14.68         20.13           94.74      684.5   
17          1        16.13         20.68          108.10      798.8   
18          1        19.81         22.15          130.00     1260.0   
19          0        13.54         14.36           87.46      566.3   

    smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0           0.11840           0.27760         0.30010              0.14710   
1           0.08474           0.07864         0.08690              0.07017   
2           0.10960           0.15990         0.19740              0.12790   
3           0.14250           0.28390         0.24140              0.10520   
4           0.10030           0.13280         0.19800              0.10430   
5           0.12780           0.17000         0.15780              0.08089   
6           0.09463           0.10900         0.11270              0.07400   
7           0.11890           0.16450         0.09366              0.05985   
8           0.12730           0.19320         0.18590              0.09353   
9           0.11860           0.23960         0.22730              0.08543   
10          0.08206           0.06669         0.03299              0.03323   
11          0.09710           0.12920         0.09954              0.06606   
12          0.09740           0.24580         0.20650              0.11180   
13          0.08401           0.10020         0.09938              0.05364   
14          0.11310           0.22930         0.21280              0.08025   
15          0.11390           0.15950         0.16390              0.07364   
16          0.09867           0.07200         0.07395              0.05259   
17          0.11700           0.20220         0.17220              0.10280   
18          0.09831           0.10270         0.14790              0.09498   
19          0.09779           0.08129         0.06664              0.04781   

    symmetry_mean  ...  radius_worst  texture_worst  perimeter_worst  \
0          0.2419  ...         25.38          17.33           184.60   
1          0.1812  ...         24.99          23.41           158.80   
2          0.2069  ...         23.57          25.53           152.50   
3          0.2597  ...         14.91          26.50            98.87   
4          0.1809  ...         22.54          16.67           152.20   
5          0.2087  ...         15.47          23.75           103.40   
6          0.1794  ...         22.88          27.66           153.20   
7          0.2196  ...         17.06          28.14           110.60   
8          0.2350  ...         15.49          30.73           106.20   
9          0.2030  ...         15.09          40.68            97.65   
10         0.1528  ...         19.19          33.88           123.80   
11       

## Standardize all column

In [ ]:
# Standardize the numeric columns
std = StandardScaler()
# Standardize all numeric columns
df[df.columns[1:]] = std.fit_transform(df[df.columns[1:]])

df.head() #verify

## Perform Stratified Split

In [63]:
# Split data
from sklearn.model_selection import train_test_split

# Split data training dan dan lainnya
# data lainnya, akan kita split lagi menjadi validasi dan testing.
# Rasio yang akan kita gunakan adalah 8:1:1
df_train, df_test = train_test_split(df, test_size=0.2, random_state=0, stratify=df['diagnosis'])

# Cek masing-masing ukuran data

print(f'Jumlah label data asli:\n{df.diagnosis.value_counts()}')
print(f'Jumlah label data train:\n{df_train.diagnosis.value_counts()}')
print(f'Jumlah label data test:\n{df_test.diagnosis.value_counts()}')

Jumlah label data asli:
diagnosis
0    357
1    212
Name: count, dtype: int64
Jumlah label data train:
diagnosis
0    285
1    170
Name: count, dtype: int64
Jumlah label data test:
diagnosis
0    72
1    42
Name: count, dtype: int64
